### PR TITLE
test(java): add BouncyCastle digest edge-case coverage

### PR DESCRIPTION
**Pull request by sachin9058**
This PR adds explicit regression coverage for additional BouncyCastle digest implementations that were already registered through `BcDigests` but did not have dedicated tests.

### Added coverage

* `WhirlpoolDigest`
* `TigerDigest`
* `RIPEMD160Digest`

### What changed

* Added `BcDigestEdgeCasesTestFile` with digest instantiations and detection assertions
* Added `BcDigestEdgeCasesTest`
* Verified detection and translation metadata for:

  * digest functionality
  * digest size
  * block size (where applicable)
  * iteration information (where applicable)

### Validation

```bash
mvn -pl java -Dtest=BcDigestEdgeCasesTest test
mvn test -pl java
```

### Notes

The digest implementations were already registered in `BcDigests`, but explicit edge-case coverage for these variants was missing. This PR focuses only on regression coverage and does not modify existing translation behavior.

Related: #31

### CODE REVIEW

**Comment by sachin9058** [comment]
@san-zrl Hi! I continued investigating BouncyCastle edge cases and opened a follow-up PR adding explicit regression coverage for additional digest implementations already registered through `BcDigests`.

Covered digests:

* WhirlpoolDigest
* TigerDigest
* RIPEMD160Digest

The focus is on improving explicit edge-case coverage and validating translation metadata without changing existing behavior.

File: java/src/test/java/com/ibm/plugin/rules/detection/bc/signer/BcDigestEdgeCasesTest.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            assertThat(digestNode.getChildren()).isEmpty();
+            assertThat(digestNode.asString()).isEqualTo("DIGEST");
+
+            // DigestSize under MessageDigest
+            INode digestSizeNode = messageDigestNode.getChildren().get(DigestSize.class);
+            assertThat(digestSizeNode).isNotNull();
+            assertThat(digestSizeNode.getChildren()).isEmpty();
+            assertThat(digestSizeNode.asString()).isEqualTo("160");
+        }
+    }
```
[the remark this answers is not in the document]
Comment (sachin9058) [reply java/src/test/java/com/ibm/plugin/rules/detection/bc/signer/BcDigestEdgeCasesTest.java:188]: Addressed the unexpected finding case by adding a final failure branch for unsupported `findingId` values. I kept the assertion structure explicit for now to stay aligned with the existing regression test style used across the repository.
